File: app/src/main/java/com/maherhanna/cheeta/PlayerPiece.java
package com.maherhanna.cheeta;

import java.util.ArrayList;

public class PlayerPiece {
    private Piece piece;
    private ArrayList<Integer> moves;

    public PlayerPiece(Piece piece,int position){
        this.piece = new Piece(piece);
        moves = new ArrayList<>();
    }

    public PlayerPiece(Piece.Type type, Piece.Color color, int position){
        this.piece = new Piece(type,color,position);
    }

    public void moveTo(int newPosition){
        this.piece.position = newPosition;
    }

    public Piece getPiece(){return this.piece;}

    public int getPosition(){
        return this.piece.position;
    }
    public int getFile(){return ChessBoard.GetFile(piece.position);}
    public int getRank(){return ChessBoard.GetRank(piece.position);}


}

File: app/src/main/java/com/maherhanna/cheeta/PlayerPiece.kt
package com.maherhanna.cheeta

public class PlayerPiece(position:Int) {
    //position in chessboard array with values from 0 to 63
    private var position:Int = position
        set(value){
        if(value < 0 || value > 63)
            throw IndexOutOfBoundsException("Given position of piece is out of board range")
        else
            field = value
    }
}
